src/extension.ts

```diff
@@ -138,7 +138,12 @@ export async function activate(ctx: vscode.ExtensionContext) {
     conf.update("default.compileCommands", compileCommandsFile, vscode.ConfigurationTarget.Workspace);
 
     const conf2 = vscode.workspace.getConfiguration("clangd");
-    conf2.update("arguments", [`--compile-commands-dir=${buildDir}`], vscode.ConfigurationTarget.Workspace);
+    conf2.update("arguments", [`--compile-commands-dir=${buildDir}`], vscode.ConfigurationTarget.Workspace).then(
+      () => {
+        vscode.commands.executeCommand("clangd.restart");
+      },
+      () => {},
```

Comment:
I don't think you need the second argument.

Reply:
This was also my initial assumption, but leaving this callback undefined causes an exception to be thrown.

Reply:
I'm not sure we should workaround that here, that's a bug that should be fixed in clangd. Such workarounds rarely get removed when they are not needed anymore, and the code becomes more bloated over time.

Reply:
I agree in principle, but we need to find a way to make it work for our users. Issuing VS Code commands by extensions is encouraged in the API docs, and I believe it is a better solution than requiring users to manually run the restart command every time the project root changes on our end.

Reply:
You can at least remove the braces from the first anonymous function.

Reply:
Done.

Reply:
Can we also get rid of conf2? Make conf a mutable variable and reuse it.

Reply:
I changed `conf` to a global configuration scope, so we only need to get it once.